### ISSUE TITLE
Tool to undo comment vote spamming

### ISSUE THREAD

**Issue by nonamethanks** [danbooru/danbooru#4640]
There should be a way to mass revert an user's comment votes. Maybe in the ban page, because look at this:
https://danbooru.donmai.us/comment_votes?commit=Search&search%5Buser_name%5D=NoBlueLives
https://danbooru.donmai.us/comment_votes?commit=Search&search%5Buser_name%5D=Cock%26Ball_Destroyer
https://danbooru.donmai.us/comment_votes?commit=Search&search%5Buser_name%5D=Admiral_Fagg0tPants

All same user using sockpuppets. There's more. Basically someone can keep creating new accounts and mass downvoting stuff into oblivion and there's no way to revert this, because all it takes is a member account to do so.

**Comment by evazion** [comment]
This is how it works:

* Comment votes now have an `is_deleted` flag.
* Comment votes are soft deleted instead of hard deleted. This includes when a user removes their own vote.
* Admins can delete comment votes by other users. This is done by going to the comment vote list and clicking the Remove button on each one.
* Deleting a vote leaves a mod action.

I'm wary about giving this to moderators because I don't want mods deleting votes unless it's to undo mass spam by sockpuppet accounts. Imagine you were on Reddit or some other site and you knew moderators could silently manipulate comment scores by removing votes if people voted the "wrong" way. It would reduce trust in the site, even if it weren't being abused.

**Comment by evazion** [comment]
Allowing mods to remove all of someone's votes when they permaban someone might be acceptable. That's harder to misuse than unrestricted vote removal. I'll leave this issue open for that.

**Comment by nonamethanks** [comment]
Yeah, allowing it to be done only on account ban is definitely an acceptable restriction.

**Comment by ghost** [comment]
Was this fixed with b5ba14bb04a919efa1143486221108e4110165dd or is something additional required?
